src/azure-cli/azure/cli/command_modules/util/custom.py

```diff
@@ -130,19 +136,37 @@ def upgrade_version(cmd, update_all=None, yes=None):  # pylint: disable=too-many
         else:
             logger.warning(UPGRADE_MSG)
     if exit_code:
-        telemetry.set_failure("CLI upgrade failed.")
+        err_msg = "CLI upgrade failed."
+        logger.warning(err_msg)
+        telemetry.set_failure(err_msg)
         sys.exit(exit_code)
-    # Updating Azure CLI and extension together is not supported in homebrewe package.
+
+    import azure.cli.core
+    import importlib
+    importlib.reload(azure.cli.core)
+    new_version = azure.cli.core.__version__
+
+    if update_cli and new_version == local_version:
+        err_msg = "CLI upgrade failed or aborted."
+        logger.warning(err_msg)
+        telemetry.set_failure(err_msg)
+        sys.exit(1)
+
+    # Python is reinstalled in another versioned directory with Homebrew, subprocess needs to be reloaded
     if installer == 'HOMEBREW' and exts:
-        logger.warning("Please rerun 'az upgrade' to update all extensions.")
-    else:
-        for ext_name in exts:
-            try:
-                logger.warning("Checking update for %s", ext_name)
-                subprocess.call(['az', 'extension', 'update', '-n', ext_name],
-                                shell=platform.system() == 'Windows')
-            except Exception as ex:  # pylint: disable=broad-except
-                msg = "Extension {} update failed during az upgrade. {}".format(ext_name, str(ex))
-                raise CLIError(msg)
-
-    logger.warning("Upgrade finished.")
+        importlib.reload(subprocess)
+
+    if exts:
+        logger.warning("Upgrading extensions")
+    for ext_name in exts:
+        try:
+            logger.warning("Checking update for %s", ext_name)
+            subprocess.call(['az', 'extension', 'update', '-n', ext_name],
+                            shell=platform.system() == 'Windows')
```

Comment:
How the subprocess could be called in homebrew?
It seems that you will call subprocess in homebrew with shell=False, right?

Reply:
Right, on macOS, shell=False should be used to call subprocess for the homebrew command to run correctly.